scripts/leapfrog/pre_redeploy.sh

```diff
@@ -112,3 +112,11 @@ popd
 pushd ${RPCO_DEFAULT_FOLDER}/rpcd/playbooks
     openstack-ansible configure-apt-sources.yml
 popd
+if [[ ! -f "${UPGRADE_LEAP_MARKER_FOLDER}/cleanup-haproxy.complete" ]]; then
+    # Ad-hoc commands to cleanup haproxy config dir.
+    ansible -m shell -a "(mkdir /root/haproxy-leap/ && cp -r /etc/haproxy/ /root/haproxy-leap/) || true" haproxy -i /opt/rpc-openstack/openstack-ansible/playbooks/inventory
+    ansible -m shell -a "cp /etc/init.d/haproxy /root/haproxy-leap/haproxy.kilo || true" haproxy -i /opt/rpc-openstack/openstack-ansible/playbooks/inventory
+    ansible -m shell -a "rm -f /etc/haproxy/conf.d/* || true" haproxy -i /opt/rpc-openstack/openstack-ansible/playbooks/inventory
+    ansible -m shell -a "cp /etc/init.d/haproxy.dpkg-dist /etc/init.d/haproxy || rm /etc/init.d/haproxy || true" haproxy -i /opt/rpc-openstack/openstack-ansible/playbooks/inventory
```

Comment:
Yes, if you don't have a backup and therefore can't restore, still remove it.
If you re-run, and don't have the file anymore, don't fail.